### PR TITLE
Add Patches tab

### PR DESCRIPTION
**Pull request by acizmarik**
This PR implements the core functionality for milestone 4.

* There is now a "Patches" tab.
* Patches tab populates automatically on process attach. Available patches are loaded by default from AppData folder.
   * If patch metadata were matched against process metadata, then user can browse them (similarly like in Metadata Browser).
   * If patch metadata were not matched, then user will see error(s) that describe what was not found.
* Each tab can be right-clicked. On right-click, a flyout menu will be shown with an "Apply" option.
   * On successful apply, user will see a messagebox with confirmation.
   * On unsuccessful apply, user will see an error with description why patch applying failed.

Additional changes:
* There is now a status bar for process attach status.
* When user creates a workspace for a new patch, they will be provided with an input dialog to enter its name (this will be translated into patch assembly name and it can help to distinguish them in "Patches" tab).
* It is now possible to have multiple method patches per assembly.
* Metadata to describe patches are no longer assembly-level but instead can be applied on individual patch classes.
   * Attributes will be checked by patches loader to determine what patches can be applied to what process
   
image

### CODE REVIEW

File: src/Meditation.UI/Controllers/PatchProcessController.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                await DialogUtilities.ShowUnhandledExceptionMessageBox(exception);
+            }
+        }
+
+        [RelayCommand]
+        public Task ReversePatch(string patchAssemblyPath, CancellationToken ct)
+        {
+            try
+            {
+                // TODO: implement patch reverse
```
Comment (LunarWhisper) [inline src/Meditation.UI/Controllers/PatchProcessController.cs:55]: [ToFix] Issue number

File: src/Meditation.UI/Services/Dialogs/DialogLifetime.cs [hunk 2]
```diff
@@ -8,36 +9,42 @@ public class DialogLifetime : IDisposable
...
+            _taskCompletionSource = new TaskCompletionSource();
             _isShown = true;
+            return _taskCompletionSource.Task;
         }
 
         public void Close()
         {
-            if (!_isShown)
-                throw new InvalidOperationException("Dialog is not opened");
+            if (!_isShown || _taskCompletionSource == null)
```
Comment (LunarWhisper) [inline src/Meditation.UI/Services/Dialogs/DialogLifetime.cs:35]: [Note] Typically, methods like `.Close` or `.Delete` are written in such a way that they do not throw an error if the entity is already closed/deleted. This makes them easier to use in `try-finally` blocks and reduces the number of fails in situations where nothing bad happened.

File: src/Meditation.UI/Services/Patches/PatchViewModelBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+            foreach (var patch in patches)
+                childrenBuilder.Add(BuildMethodViewModel(typeMetadataEntry, patch));
+
+            return new PatchTypeMetadataEntry(typeMetadataEntry.Name, typeMetadataEntry.FullName, childrenBuilder.ToImmutable());
+        }
+
+        private MetadataEntryBase BuildMethodViewModel(TypeMetadataEntry typeMetadataEntry, PatchInfo patch)
+        {
+            var targetMethod = typeMetadataEntry.Children
```
Comment (LunarWhisper) [inline src/Meditation.UI/Services/Patches/PatchViewModelBuilder.cs:69]: [ToFix] Some kind of complex logic in non-trivial LINQ expression with unsafe casts. Fix it please.

**Review by LunarWhisper — CHANGES_REQUESTED** [review]
Looks good. Few medium issues.

File: src/Meditation.UI/Controllers/PatchProcessController.cs (hunk 1 above)
Comment (acizmarik) [reply src/Meditation.UI/Controllers/PatchProcessController.cs:55]: This is a temporary TODO. It is already implemented here: #54

File: src/Meditation.UI/Services/Patches/PatchViewModelBuilder.cs (hunk 3 above)
Comment (acizmarik) [reply src/Meditation.UI/Services/Patches/PatchViewModelBuilder.cs:69]: The cast is not unsafe (they are filtered first). But I will simplify it.
```csharp
typeMetadataEntry.Children
                .Where(c => c is MethodMetadataEntry)
                .Cast<MethodMetadataEntry>()
```

**Review by LunarWhisper — APPROVED** [review]
LGTM

**Comment by acizmarik** [comment]
After agreement with @aleksylum, I am merging

**Pull request merged**